### DISCUSSION TITLE
How can I leverage `@next/font` to create a performant font picker?

### DISCUSSION THREAD

**Discussion by jsefiani** [vercel/next.js#43411] [Help]
Hi everyone,

I'm looking for ideas on how to build a performant font picker using `@next/font`. 

Is this even possible?

An obvious naive implementation would be to import all fonts from `@next/font/google`, call each font function and use it to create an array of fonts.

Something like this:
```
import { Akshar } from "@next/font/google";

const akshar = Akshar();

export const fonts = [{label: "Akshar", value: "akshar", font: akshar}];
```

This works fine if you have a couple of fonts, but it becomes a performance bottleneck when you want to display hundreds of fonts because Next loads them all at once.

Does anyone have a better solution?

**Comment by parweb** [comment]
i did it not performant at all :/
looking for fresh idea
i find this, didn't test yet but seem legit
https://medium.com/@made4uo/how-to-add-font-picker-in-next-js-8bebfadf6795
